ui/client/components/AddProcessDialog.tsx

```diff
@@ -1,62 +1,72 @@
 import HttpService from "../http/HttpService"
 import "../stylesheets/visualization.styl"
 import ValidationLabels from "./modals/ValidationLabels"
 import * as DialogMessages from "../common/DialogMessages"
+import {showProcess} from "../actions/nk/showProcess"
 
-//TODO: Consider integrating with GenericModalDialog
-class AddProcessDialog extends React.Component {
-
-  static propTypes = {
-    categories: PropTypes.array.isRequired,
-    isOpen: PropTypes.bool.isRequired,
-    onClose: PropTypes.func.isRequired,
-    isSubprocess: PropTypes.bool,
-    visualizationPath: PropTypes.string.isRequired,
-    message: PropTypes.string.isRequired,
-    clashedNames: PropTypes.array,
-  }
+type State = {
+  processId: string,
+  processCategory: string,
+}
 
-  initialState(props) {
-    return {processId: "", processCategory: _.head(props.categories) || ""}
+//TODO: move this validation to backend to simplify FE code
+function getAlreadyExistsValidator(clashedNames): Validator {
+  return {
+    isValid: (name) => !clashedNames.includes(name),
+    message: DialogMessages.valueAlreadyTaken,
+    description: DialogMessages.valueAlreadyTakenDescription,
+    validatorType: ValidatorType.Frontend,
+    handledErrorType: HandledErrorType.AlreadyExists,
   }
+}
 
-  constructor(props) {
-    super(props)
-    this.state = this.initialState(props)
-  }
+function prepareNameValidators(clashedNames) {
+  const alreadyExistsValidator = getAlreadyExistsValidator(clashedNames)
+  return [
+    mandatoryValueValidator,
+    alreadyExistsValidator,
+  ]
+}
+
+class AddProcessDialog extends React.Component<Props, State> {
+  initialState = () => ({processId: "", processCategory: this.props.categories[0] || ""})
+
+  state = this.initialState()
 
   closeDialog = () => {
-    this.setState(this.initialState(this.props))
+    this.setState(this.initialState())
     this.props.onClose()
   }
 
-  confirm = () => {
-    const processId = this.state.processId
-    HttpService.createProcess(this.state.processId, this.state.processCategory, this.props.isSubprocess).then((response) => {
-      this.closeDialog()
-      history.push(VisualizationUrl.visualizationUrl(processId))
-    })
+  confirm = async () => {
+    const {processCategory, processId} = this.state
+    await this.createProcess(processId, processCategory)
+    this.closeDialog()
+  }
+
+  private async createProcess(processId: string, processCategory: string) {
+    const {isSubprocess, showProcess} = this.props
+    await HttpService.createProcess(processId, processCategory, isSubprocess)
+    showProcess(processId)
   }
 
   render() {
-    const titleStyles = EspModalStyles.headerStyles("#2d8e54", "white")
-    const nameValidators = prepareNameValidators(this.props.clashedNames)
+    const {message, clashedNames, categories, isOpen} = this.props
+    const {processId} = this.state
+
+    const titleStyles = EspModalStyles.headerStyles("#2D8E54", "white")
```

Comment:
Please change `EspModalStyles` to `NkModalStyles`.

Reply:
fixed in [6]